core/src/test/scala/org/apache/spark/scheduler/ReplayListenerSuite.scala

```diff
@@ -151,7 +153,10 @@ class ReplayListenerSuite extends SparkFunSuite with BeforeAndAfter with LocalSp
    * assumption that the event logging behavior is correct (tested in a separate suite).
    */
   private def testApplicationReplay(codecName: Option[String] = None) {
-    val logDirPath = Utils.getFilePath(testDir, "test-replay")
+    val logDir = new File(testDir.getAbsolutePath, "test-replay")
+    // Here, it creates `Path` from the URI instead of the absolute path for the explicit file
+    // scheme so that the string representation of this `Path` has leading file scheme correctly.
+    val logDirPath = new Path(logDir.toURI)
```

Comment:
@HyukjinKwon I think this change it self looks reasonable.
`resolveURI` should seems to be fixed so that Windows' path is handled correctly. 
If  `C:/path/to/some/file` is passed to `resolveURI`, the letter drive "C" should not parsed as URI scheme.